File: attachment-microservice/src/main/scala/data/service/AttachmentService.scala
package data.service

import java.io.File

import data.DTO.AttachmentData
import data.entity.Attachment
import org.springframework.web.multipart.MultipartFile

/**
  * attachment service
  *
  * @author 刘佳兴
  * @version V1.0
  */
trait AttachmentService {

  /**
    * save attachment and storage file
    *
    * @param multipartFile MultipartFile
    * @param relativePath  相对存储路径
    * @return
    */
  def save(multipartFile: MultipartFile, relativePath: String): Attachment


  def get(name: String): File

}

File: attachment-microservice/src/main/java/data/service/AttachmentService.java
package data.service;

import data.entity.Attachment;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.io.InputStream;

/**
 * 文件信服务类
 *
 * @author 刘佳兴
 */
public interface AttachmentService {

    /**
     * save attachment and storage file
     *
     * @param inputStream  文件上传流
     * @param attachment   附件对象
     * @return Attachment
     */
    Attachment save(InputStream inputStream, Attachment attachment);

    /**
     * soft remove by name
     *
     * @param name uuid
     * @return Attachment
     */
    Attachment remove(String name);

    /**
     * get attachment by name
     *
     * @param name uuid
     * @return Attachment
     */
    Attachment get(String name);

    /**
     * 读取文件
     *
     * @param name uuid
     * @return 读取文件流
     */
    InputStream read(String name);

}
